src/eca/features/tools.clj

```diff
@@ -157,38 +165,41 @@
 
 (defn approval
   "Return the approval keyword for the specific tool call: ask, allow or deny."
-  [all-tools tool-call-name args db config]
-  (let [{:keys [server require-approval-fn]} (first (filter #(= tool-call-name (:name %))
-                                                            all-tools))
-        {:keys [allow ask deny byDefault]} (get-in config [:toolCall :approval])]
-    (cond
-      (and require-approval-fn (require-approval-fn args {:db db}))
-      :ask
+  ([all-tools tool-call-name args db config]
+   (approval all-tools tool-call-name args db config nil))
```

Comment:
Why keep this arity? is there a place where we don't have behavior or makes sense to ignore it?